File: anagram.js

function sortChars(word) {
	var charArr = word.split('');
	var key = charArr.sort().join('');
	return key;
}

var fileName = 'words';
var fs = require('fs');
fs.readFile(fileName, 'utf8', function(err, data) {
	if(!err) {
		var anagramHash = {};
		var wordArr = data.toString().split('\n');
		for(var i in wordArr) {
			var word = wordArr[i];
			if(word) {
				key = sortChars(word);
				if((key in anagramHash) == false) {
					anagramHash[key] = [];
				}
				anagramHash[key].push(word);
			}
		}
		for(var x in anagramHash) {
			if(x) {
				var a = anagramHash[x];
				if(a.length > 3) {
					console.log(x + '\t=>\t' + a.join(', '));
				}
			}
		}
	} else {
		throw err;
	}
});

File: anagram.py
from collections import defaultdict

f = open('words')
words = f.readlines()
f.close()

anaDict = defaultdict(list)

for word in words:
	word = word.rstrip()
	l = list(word)
	l.sort()
	key = ''.join(l)
	anaDict[key].append(word)

for key, val in anaDict.items():
    if len(val) > 3:
		   print(key + ' => ' + ', '.join(val))
